### PR TITLE
Ensure Collections calculate MinSize consistently

### PR DESCRIPTION
**Pull request by stuartmscott**
### Description:
All three collections will not return the larger of a element's minsize and the scroller's minsize.

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Review by okratitan — APPROVED** [review]

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Good change, but I think there is a test missing for this.
Possibly just Table?

**Comment by stuartmscott** [comment]
Tests added

**Review by okratitan — DISMISSED** [review]

File: widget/table.go [hunk 1]
```diff
@@ -244,7 +244,7 @@ func (t *tableRenderer) Layout(s fyne.Size) {
 }
 
 func (t *tableRenderer) MinSize() fyne.Size {
-	return t.cellSize
+	return t.t.scroll.MinSize().Max(t.cellSize.Add(fyne.NewSize(theme.Padding(), 0)))
```
Comment (andydotxyz) [inline widget/table.go:247]: If we are leaving space for the indicators then table needs to have the same on the vertical direction

File: widget/list.go [hunk 2]
```diff
@@ -213,7 +213,7 @@ func (l *listRenderer) Layout(size fyne.Size) {
 }
 
 func (l *listRenderer) MinSize() fyne.Size {
-	return l.scroller.MinSize()
+	return l.scroller.MinSize().Max(l.list.itemMin)
```
Comment (andydotxyz) [inline widget/list.go:216]: Does itemMin include the padding amount for the indicator?

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Following discussions in Slack I don't think we're leaving enough space for each of them to show indicators and a full cell/item/node on screen at minimum size.

The one I am sure about I commented inline - table needs to have padding at the top as well

File: widget/list.go (hunk 2 above)
Comment (okratitan) [reply widget/list.go:216]: Yes

Comment (stuartmscott) [reply widget/list.go:216]: Yup

File: widget/table.go (hunk 1 above)
Comment (stuartmscott) [reply widget/table.go]: Done

**Comment by stuartmscott** [comment]
Added padding for column selection indicator as well

**Review by okratitan — APPROVED** [review]

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**